src/artifacts_keyring/plugin.py

```diff
@@ -45,7 +45,9 @@ def __init__(self):
                 # Ensure the plugins directory executable permissions are set so Python can execute
                 # the Credential Provider plugin.
                 try:
-                    os.chmod(tool_path_root, 0o755)
+                    tool_path = os.path.join(tool_path_root, 'CredentialProvider.Microsoft')
```

Comment:
Was this also tested on Linux? I wonder if we'll need to split it for each OS

Reply:
yes!